doc/adr/0002-use-pact-for-contract-testing.md

```diff
@@ -0,0 +1,59 @@
+# 2. Use Pact for contract testing
+
+Date: 2023-05-24
+
+## Status
+
+Accepted
+
+## Context
+
+We're developing a service with a frontend client (the UI) that communicates
+with a separate API, written by different developers often working in parallel.
+When running integration tests, the UI team do this against a mocked version of
+the API which may not always be in the state we expect.
+
+We need to ensure that any changes to the API don't break existing
+functionality in the user interface. Although we could use Cypress to test the
+deployed application, this would have a number of drawbacks:
+
+- We have to maintain integration and e2e tests, which are functionally
+  similar, but different enough for there to be differences in implementation
+- We have to wait for a new version of the code to be deployed before running
+  our e2e tests, meaning the feedback loop for noticing a problem and fixing it
+  is long
+- e2e tests are slow running, slowing down our build pipeline
+- As the tests run in a real environment, we are filling the database with test
+  data, which has to be cleaned out
+- The tests can be flaky, as pass for reasons that are outside our control,
+  such as upstream APIs being down, or performance issues
+- When tests do fail, it's not always clear _what_ has broken, and it can be
+  hard to track down the source of the failure
+
+## Decision
+
+With all this in mind, we've decided to use [Pact](https://pact.io/) to carry
+out Contract Testing when testing our API client classes.
+
+When running Pact contract tests, JSON files called "Pacts" are generated by
+"the consumer" (our UI). These are published to the Pact Broker on a successful
+CI build. "The provider" then pulls these contracts and verifies its own
+response shape against them to ensure it's sending what's expected to the UI.
+If the shape is different to what the consumer expects, the build will fail and
+prevent us merging anything that could break functionality.
+
+## Consequences
+
+This will hopefully speed up our development, and raise any deviation in the
+contract between the API and the UI application in a timely manner without the
+need for expensive and flaky E2E tests. When API changes do cause the tests to
```

Comment:
`e2e` is now `E2E` - I think it's worth spelling out in at least the first instance, and then either using a provided abbreviation or continuing to spell it out

Reply:
Yeah I've spelt this out above and made it consistent throughout.